### PR TITLE
Disable sigaltstack on Apple platforms

### PR DESCRIPTION
**Pull request by keith**
According to https://reviews.llvm.org/D28265 using `sigaltstack` with
`backtrace()` (which absiel uses internally to get stack trace info) is
not compatible. Using them together results in crashes in tests not
printing backtraces only on macOS

https://github.com/envoyproxy/envoy-mobile/issues/1827

### CODE REVIEW

File: source/common/signal/signal_action.cc [hunk 1]
```diff
@@ -54,12 +54,14 @@ void SignalAction::sigHandler(int sig, siginfo_t* info, void* context) {
 }
 
 void SignalAction::installSigHandlers() {
+#if !defined(__APPLE__)
```
Comment (mattklein123) [inline source/common/signal/signal_action.cc:57]: Can you add comments on why this is disabled for apple?

**Review by mattklein123 — COMMENTED** [review]
Awesome thanks for fixing. cc @rojkov who is probably our best bet for a reviewer who understands all of this. :)

**Comment by keith** [comment]
a 404 from brew 🤔

**Review by rojkov — APPROVED** [review]
Thanks for fixing! The consequence is that stack overflows become difficult to debug on Apple, but given sigaltstack has been already non-functional on the platform it's better to go this way and to get meaningful backtraces for other cases at least.

**Review by ggreenway — APPROVED** [review]

**Comment by keith** [comment]
Thanks!

**Pull request merged**